### DISCUSSION TITLE
Open multiple documents

### DISCUSSION THREAD

**Discussion by PauLuke** [ahrm/sioyek#268] [Q&A] [answered]
When I open a new document, it closes the later. How can I solve this?

**Comment by ahrm** [comment, marked as answer]
If you are using the command line, you can pass the `--new-instance` option to open a new sioyek instance. Alternatively, you can set `new-instance 1` in your `prefs_user.config` to make this is default behaviour.
See https://sioyek-documentation.readthedocs.io/en/latest/configuration.html?highlight=new-instance#should-launch-new-instance.

Also in new version of sioyek we have added a `--new-window` command line option as well as a `new_window` command within sioyek itself which is more efficient than `new-instance`. It is not available in `1.3.0` release though, so you have to compile it yourself or use the preview builds here:
https://github.com/hexomancer/sioyek/releases/tag/v0.31.336

**Reply by PauLuke** [reply]
Now I get it. Thanks for the reply and also for the amazing program.

**Reply by kasturikandalam** [reply]
this doesn't seem to work on windows. The file just gets replaced

**Comment by andriandreo** [comment]
Thus, regarding the default configuration, new documents should be opened as new windows by default, right? This is not happening in my case (Arch Linux, sioyek 2.0.0, sioyek AUR pkg). Thanks!

**Reply by ahrm** [reply]
No, default configuration reuses the same window.

**Reply by andriandreo** [reply]
Thanks for the quick reply and the great software! So... should I try sth like `should_launch_new_window 1` (because I tried and that did not work either...)

**Reply by ahrm** [reply]
Yes. If it doesn't work it is either a bug or maybe your configuration is not applied? Are you sure your config file is being read (e.g. are other settings working if you change them in your config file). Also does passing `--new-window` in command line work?

**Reply by andriandreo** [reply]
`--new-window` in command line does work. But nothing regarding `should_launch_new_window 1` inside `prefs_user.config` (the rest of configs such as custom highlighting are indeed applied).

**Reply by ahrm** [reply]
Does the issue happen with appimage builds in the release section of this repo?

**Reply by andriandreo** [reply]
Yes, the issue is happening for both AUR pkg and AppImage in Release v2.0.0

**Reply by davdent** [reply]
I have a very similar issue running Sioyek 2.0 on Ubuntu 23.10.  Setting ``should_launch_new_window 1`` inside pref_user.config does not work. And in my case, passing  --new-window on the command line does _not_ work.  Also, neither passing --new-instance on the command line, nor setting `should_launch_new_instance 1` work for me.  When I try to pass --new-instance on the command line, I get the message "Unknown option 'new-instance". However, other configs set in `prefs_user.config` are being applied as expected.

**Reply by jackvhughes** [reply]
having the same issue as above with MacOS Sonoma 14.5 M2

**Comment by RYQ-22** [comment]
Same. `shold_launch_new_window 1` doesn't work with the `AppImage` version.

**Reply by lefthairynut** [reply]
Any update on this issue? Having the same problem and it's weird it's not being answered by the main author.

**Reply by ahrm** [reply]
Does the issue happen in the experimental builds here: https://github.com/ahrm/sioyek/releases/tag/sioyek3-alpha0?

**Reply by RYQ-22** [reply]
> Any update on this issue? Having the same problem and it's weird it's not being answered by the main author.

I tried to modify prefs.config instead of prefs_user.config and it worked at least for me. Maybe you can try it.

**Reply by lefthairynut** [reply]
@ahrm Sorry for the late response. Using the AppImage in the release, yes, it's still the same issue with the same config as before. The recurring problem is that it _rarely_ works. And by that I mean I can open new window with CTRL-t or `new_window` command but it occures rarely the way it should open a new window normally. most of the time, it seems like the new window appears for a fraction of a second and it disappears quickly.
 The workaround I'm currently using is to simply open a new instance using `sioyek <file-path>`.
These are the two related lines in my prefs_user.config:
```
should_launch_new_window 1
should_launch_new_instance 1
```

**Reply by lefthairynut** [reply]
> > Any update on this issue? Having the same problem and it's weird it's not being answered by the main author.
> 
> I tried to modify prefs.config instead of prefs_user.config and it worked at least for me. Maybe you can try it.

doesn't work for me.

**Reply by inmzhang** [reply]
I can confirm that `should_launch_new_window 1` does not work with `sioyek-appimage` AUR package on Arch. Then I use `sioyek` AUR package and it works well.

**Comment by WhyNotHugo** [comment]
I'm hitting the same bug on Alpine Linux Edge, with 2.0.0-r33.

I tried adding all of these to `~/.config/sioyek/prefs.config`, but none of them seemed to work:

```
new-instance 1
should_launch_new_window 1
should_launch_new_instance 1
```

**Reply by WhyNotHugo** [reply]
Okay, saving the file as `prefs_user.config` worked, despite the current documentation indicating that `prefs.config` should contain the default configuration.

However, when I press `q` to close one instance, all other instance terminate too. It seems they're interconnected in some funny way?

**Reply by WhyNotHugo** [reply]
Looks like `q` is bound to `close all windows` by default. Adding `close_window q` to `keys_user.config` allows me to close just the current window without others.
